sample-tracking-status-overview-app/src/main/groovy/life/qbic/portal/sampletracking/components/projectoverview/ProjectOverviewView.groovy

```diff
@@ -31,7 +34,7 @@ import life.qbic.portal.sampletracking.components.projectoverview.subscribe.Subs
  *
 */
 @Log4j2
-class ProjectOverviewView extends VerticalLayout{
+class ProjectOverviewView extends VerticalLayout implements VisibilityChangeListener{
```

Comment:
No need to make this implement a listener interface. As you have defined the listener as a functional interface you can simply pass a lambda or closure and do not need this dependency for the whole class. Implementing the listener itself in the view leads to a convoluted design later on since you cannot replace listeners easily.